questions/459-medium-flatten/template.ts

```diff
@@ -1 +1,5 @@
-export type Flatten = any
+export type Flatten<T> = T extends [infer First, ...infer Rest]
```

Comment:
What about defining a helper type for avoid nested ternaries?
```

type FlattenHelper<T> = T extends unknown[] ? Flatten<T> : [T];

export type Flatten<T extends unknown[]> = T extends [infer First, ...infer Rest]
  ? [...FlattenHelper<First>, ...Flatten<Rest>]
  : [];
```

Reply:
Sure, it would be a better way to get the same result but simplyfing the nested ternaries.